RTClib.cpp

```diff
@@ -347,20 +374,32 @@ void RTC_Millis::adjustDrift(int drift) { lastMillis = 1000 - drift; }
 */
 /**************************************************************************/
 DateTime RTC_Millis::now() {
-  uint32_t elapsedSeconds = (millis() - lastMillis) / millisPerSecond;
-  lastMillis += elapsedSeconds * millisPerSecond;
+  uint32_t elapsedSeconds = (millis() - lastMillis) * 1000 / microsPerSecond;
```

Comment:
The multiplication will overflow unless `now()` is called at least once every 71.6 minutes (and not 49.7 days, as stated in the comment). This is the same constraint we already have with `RTC_Micros`. Given that this constraint is the only drawback of `RTC_Micros` vs. `RTC_Millis`, this change makes `RTC_Millis` redundant: it could be replaced by an alias to `RTC_Micros`.

Reply:
Fixed by reducing `RTC_Millis` adjust drift precision to parts per thousand.

Reply:
Parts per thousand is an extremely coarse resolution for tuning a clock. Also note that, because of the way the integer division rounds, if the user types

```c++
clock.adjustDrift(+12);  // +12 ppm is about one second per day faster
```

his clock will speed up by +1,000&nbsp;ppm, or about 86&nbsp;seconds per day.

Reply:
I wasn't clear enough in my wording, it's changed to a resolution of parts per thousand but the user still inputs it in parts per million. This means that

```
RTC_Millis rtc;
rtc.adjustDrift(12);
```

Will have no effect. Please check my latest commit.

Even then, the programmer using `adjustDrift()` will be responsible for ensuring the drift adjustment is appropriate, not us; if they input some ridiculously high drift then that's their fault.

I plan on implementing ppm resolution to `RTC_Millis` in a separate pull request.

Reply:
Your wording, and your code, were quite clear.

```c++
rtc.adjustDrift(12);
```

will have the effect of setting

```c++
millisPerSecond = (1000000L - 12) / 1000;
```

The right hand side of this expression, if evaluated as real numbers, would give 999.988. However, this is an integer division, and in C++ integer division rounds towards zero, which results in setting `millisPerSecond` to 999. This is a +1,000 ppm speed correction, which is both huge and very far from what the user requested.

> the programmer using `adjustDrift()` will be responsible for ensuring the drift adjustment is appropriate, not us

Of course! But if they find, through careful calibration, that +12 ppm is the appropriate correction, they can reasonably expect that `rtc.adjustDrift(12);` will apply a correction that is not too far from what they requested. The library should avoid breaking such reasonable user expectations.

Reply:
Alright, this is clearly becoming overly-complicated for what was supposed to be a very simple pull request. For now, I've just removed the `adjustDrift()` function from `RTC_Millis`. I've wrote `adjustDrift()` back into `RTC_Super` to enforce consistency across the RTC classes.